### DISCUSSION TITLE
Can we have the source of the server please ?

### DISCUSSION THREAD

**Discussion by BararDiHoumes** [miroslavpejic85/p2p#36] [General]
Sorry for asking this way, but can we have the source of the server , so we can create our own nodes ?
Thank You in advance dear Miroslav.

**Comment by IanMcLumber** [comment]
Yes please. Using external server for something quite this dangerous is really making me scary. 
Please provide the server source.

**Comment by miroslavpejic85** [comment]
Hello @BararDiHoumes and @IanMcLumber,
P2P uses Google Stun server as default just to get endpoints in order to connect peers between them.  There is nothing dangerous :) you can change the Stun server from the app, or install your own as [coturn](https://github.com/coturn/coturn). More detail about the Stun what is it, you can found [here](https://en.m.wikipedia.org/wiki/STUN). As for the traffic and all direct between peers thanks to the direct UDT connection, there are no servers in between. The current source code is complete, the only thing if you don't want to use the default Stun, as I told you you can change it. Thank you.
